File: clojure/spec/euler/level1/problem004_spec.clj
(ns euler.level1.problem004-spec
  (:require
    [speclj.core :refer :all]
    [euler.level1.problem004 :refer :all]))

; https://projecteuler.net/problem=4

(describe "Euler Problem #4"

  (it "palindrome?"
    (should= true (palindrome? 0))
    (should= true (palindrome? 1))
    (should= true (palindrome? 5))
    (should= false (palindrome? 10))
    (should= true (palindrome? 11))
    (should= true (palindrome? 22))
    (should= false (palindrome? 23))
    (should= false (palindrome? 100))
    (should= true (palindrome? 101)))

  (it "max-palindrome-of-multiples-between"
    (should= 0 (max-palindrome-of-multiples-between 0 0))
    (should= 1 (max-palindrome-of-multiples-between 0 1))
    (should= 4 (max-palindrome-of-multiples-between 0 2))
    (should= 121 (max-palindrome-of-multiples-between 0 12))
    (should= 9009 (max-palindrome-of-multiples-between 10 99)))

  (it "Solves #4"
    (should= 9 (euler-4 1))
    (should= 9009 (euler-4 2))
    (should= 906609 (euler-4 3))))

File: ruby/spec/problem004_spec.rb
require 'problem004'

# https://projecteuler.net/index.php?section=problems&id=4

describe "Euler Problem #4" do
  it "Solves #1" do
    expect(euler_4(1)).to eq 9
    expect(euler_4(2)).to eq 9009
    expect(euler_4(3)).to eq 906609
  end

  it "palindrome?" do
    expect(palindrome?(1)).to be true
    expect(palindrome?(10)).to be false
    expect(palindrome?(11)).to be true
    expect(palindrome?(12)).to be false
    expect(palindrome?(44)).to be true
    expect(palindrome?(909)).to be true
    expect(palindrome?(1000)).to be false
    expect(palindrome?(1001)).to be true
  end
end
